Applications/Spire/Source/Ui/MoneySpinBox.cpp

```diff
@@ -0,0 +1,70 @@
+#include "Spire/Ui/MoneySpinBox.hpp"
+#include <QHBoxLayout>
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  auto to_money(RealSpinBox::Real value) {
+    if(auto money = Money::FromValue(value.str(
+        std::numeric_limits<Money>::digits10, std::ios_base::dec))) {
+      return *money;
+    }
+    throw std::runtime_error("MoneySpinBox: failed to convert Real to Money.");
+  }
+}
+
+MoneySpinBox::MoneySpinBox(Money value, QWidget* parent)
+    : QAbstractSpinBox(parent) {
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins({});
+  m_spin_box = new RealSpinBox(to_real(value), this);
+  m_spin_box->set_minimum_decimal_places(2);
+  // TODO: workaround for Money::CENT compiling as 0.
+  set_step(Money(0.01));
```

Comment:
Don't use binary floating point values unless they can be represented exactly. There is no way to represent the value ```0.01``` using a ```double```.

The best thing to do is understand why this is the case, and then when you really understand how a ```double``` works, use a site like this to determine if a value has an exact floating point binary representation:

https://www.binaryconvert.com/convert_double.html

Reply:
Also, update your Visual Studio to the latest version and see if this bug still exists.